bdwgraph.cpp

```diff
@@ -135,3 +135,77 @@ void BDWGraph::drawZeroAxis()
     
     qDebug() << "BDW zero axis drawn at x:" << zeroPoint.x();
 }
+
+/**
+ * @brief Override drawDataLine to use dashed lines for BDW graph
+ *
+ */
+void BDWGraph::drawDataLine(const QString &seriesLabel, bool plotPoints)
+{
+    if (!graphicsScene || !dataSource || dataSource->isEmpty() || !dataRangesValid)
+    {
+        return;
+    }
+
+    const auto &yData = dataSource->getYDataSeries(seriesLabel);
+    const auto &timestamps = dataSource->getTimestampsSeries(seriesLabel);
+
+    // Filter data points to only include those within the current time range
+    std::vector<std::pair<qreal, QDateTime>> visibleData;
+    for (size_t i = 0; i < yData.size(); ++i)
+    {
+        if (timestamps[i] >= timeMin && timestamps[i] <= timeMax)
+        {
+            visibleData.push_back({yData[i], timestamps[i]});
+        }
+    }
```

Comment:
This needs to be a waterfall graph level method / currently the waterfall data will return the right data. Don't do another loop here. This needs to be handled in the underlying layer. Especially because I have some code there that figures out how to sample the data too incase we use long timespans.